extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/Operation.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc;
+
+import com.hazelcast.jet.annotation.EvolvingApi;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * Describes the nature of the event in CDC data. Equivalent to various
+ * actions that can affect a database record: insertion, update and
+ * deletion. Has some extra special values like "sync" which is just
+ * like an insert, but originates from a database snapshot (as opposed
+ * database changelog) and "unspecified" which is used for a few special
+ * CDC events, like heartbeats.
+ *
+ * @since 4.2
+ */
+@EvolvingApi
+public enum Operation {
+    /**
+     * Change event doesn't have an operation field, for example
+     * heartbeats.
+     */
+    UNSPECIFIED(null),
+    /**
+     * Just like {@link #INSERT}, but coming from the DB snapshot (as
+     * opposed to trailing the DB changelog).
+     */
+    SYNC("r"),
+    /**
+     * Record insertion, sourced from the DB changelog.
+     */
+    INSERT("c"),
+    /**
+     * Record update, sourced from the DB changelog.
+     */
+    UPDATE("u"),
+    /**
+     * Record deletion, sourced from the DB changelog.
+     */
+    DELETE("d");
+
+    private final String id;
+
+    Operation(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Parses the string constants used in CDC messages for describing
+     * operations into enum instances.
+     * <p>
+     * Null will be parsed as {@link #UNSPECIFIED}.
+     *
+     * @throws ParsingException if the input string doesn't represent
+     * an expected value.
+     */
+    public static Operation get(@Nullable String id) throws ParsingException {
+        Operation[] values = values();
```

Comment:
seems now you have to pay cost of comparison twice: once per figuring out the string mapping, and then the other conditional operation you will do for the operation. Why not just publish it as string and refer to constants?

Reply:
I've tried a couple of versions, but I couldn't find one that I liked... Considering that the calls in questions will be made only on demand and that the operation values doesn't play any part in serde either, I'm leaning towards keeping it as is, for the moment. Will give it another look tomorrow, with a fresh mind.

Reply:
What was about it that you didn't like? I think you'll be creating more garbage by creating the extra Operation objects, without adding much value. When used with the map sink for example you'll need to do conditional operations on it.

Reply:
Index based lookup in array type of solution is pushed.